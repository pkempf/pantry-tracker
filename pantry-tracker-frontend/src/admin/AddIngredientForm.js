import React, { useContext } from "react";
import { Redirect } from "react-router-dom";
import Container from "react-bootstrap/Container";
import PantryForm from "../forms/PantryForm";
import UserContext from "../UserContext";
import AlertContext from "../AlertContext";
import PantryApi from "../api";

const AddIngredientForm = () => {
  const user = useContext(UserContext);
  const { setMessage } = useContext(AlertContext);

  const formFields = [
    {
      name: "name",
      label: "Name:",
      inputType: "text",
      initialValue: "",
      placeholder: "Example: Maple syrup",
      required: true,
    },
    {
      name: "description",
      label: "Description:",
      inputType: "text",
      initialValue: "",
      placeholder: "Enter the ingredient's description here",
      required: false,
      asOverride: "textarea",
      styleOverride: { height: "100px" },
    },
    {
      name: "type",
      label: "Type:",
      inputType: "text",
      initialValue: "",
      placeholder: "Example: vegetable",
      required: false,
    },
  ];

  async function add(ingredient) {
    try {
      let newIngredient = await PantryApi.addIngredient(ingredient);
      if (newIngredient.name && newIngredient.name === ingredient.name) {
        setMessage({
          text: `Added ingredient ${newIngredient.name}!`,
          variant: "success",
        });
      }
    } catch (err) {
      setMessage({
        text: `Error adding ingredient ${ingredient.name}`,
        variant: "danger",
      });
    }
  }

  if (!user) {
    return <Redirect to="/" />;
  }

  return (
    <Container className="AddIngredientForm justify-content-center">
      <PantryForm
        formTitle={`Add Ingredient`}
        fields={formFields}
        submitButtonText="Add Ingredient"
        processData={(data) => add(data)}
        backButtonTo={user.isAdmin ? "/admin" : "/ingredients"}
      />
    </Container>
  );
};

export default AddIngredientForm;
