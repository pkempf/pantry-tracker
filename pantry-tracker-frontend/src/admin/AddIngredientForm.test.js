import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AddIngredientForm from "./AddIngredientForm";
import UserContext from "../UserContext";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <AddIngredientForm />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: "test" }}>
        <AddIngredientForm />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
