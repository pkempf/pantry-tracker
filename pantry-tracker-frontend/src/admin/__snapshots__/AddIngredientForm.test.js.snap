// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="AddIngredientForm justify-content-center container"
  >
    <div
      class="PantryForm justify-content-center row"
    >
      <div
        class="col-md-8 col-sm-10"
      >
        <div
          class="card"
        >
          <h4
            class="card-header"
          >
            Add Ingredient
          </h4>
          <div
            class="card-body"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="name"
                >
                  Name:
                </label>
                <input
                  class="form-control"
                  id="name"
                  name="name"
                  placeholder="Example: Maple syrup"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="description"
                >
                  Description:
                </label>
                <textarea
                  class="form-control"
                  id="description"
                  name="description"
                  placeholder="Enter the ingredient's description here"
                  style="height: 100px;"
                  type="text"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="type"
                >
                  Type:
                </label>
                <input
                  class="form-control"
                  id="type"
                  name="type"
                  placeholder="Example: vegetable"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="mr-2 btn btn-primary"
                type="submit"
              >
                Add Ingredient
              </button>
              <a
                class="btn btn-secondary"
                href="/ingredients"
              >
                Back
              </a>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
